File: String Compression.js
var compress = function(chars) {
    if(chars.length < 2) return chars.length;
    var start = 0;
    var end = start;
    var count = 0;
    while(end <= chars.length){
        
        if(chars[start] !== chars[end]){
            if(count !== 1){
                count = count.toString();
                for(let i=0; i<count.length; i++){
                    chars[++start] = count[i];
                }
            } 
            chars[++start] = chars[end];
            count = 0;
        }
        count++;
        end++;
    }
    return start;
};

File: String Compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        write, read = 0, 0
        while read < len(chars):
            count = 1
            while read < len(chars)-1 and chars[read] == chars[read+1]:
                count += 1
                read += 1
            chars[write] = chars[read]
            if count > 1:
                count = str(count)
                for digit in count:
                    write += 1
                    chars[write] = digit
            
            write += 1
            read += 1
            
        return write
